feat(gitlab): implement merge_after for automergeSchedule

dominikfoerster2003 opened the pull request:
<!-- If this is your first pull request: sign the CLA with this GitHub app: https://cla-assistant.io/renovatebot/renovate -->
<!-- Make sure the `Allow edits and access to secrets by maintainers` checkbox is checked on this pull request. -->
<!-- Please read https://github.com/renovatebot/renovate/blob/main/.github/contributing.md before you create your pull request.-->

## Changes
- Add automergeSchedule and timezone to PlatformPrOptions interface
- Extend getPlatformPrOptions to pass schedule configuration to platforms
- Modify GitLab tryPrAutomerge to set merge_after when outside schedule
- Use existing getNextScheduleTime and isScheduledNow functions
- Support both Cron and Later.js schedule syntax with timezone awareness
- Add comprehensive documentation for the new functionality
<!-- Describe what behavior is changed by this PR. -->

## Context

- Closes #37028
<!-- Describe why you're making these changes if it's not already explained in a corresponding issue. -->
<!-- If you're closing an existing issue with this pull request, use the keyword Closes #issue_number. -->
<!-- If you're referencing an issue with this pull request, put it in a Markdown list like this: - #issue_number. -->

## Documentation (please check one with an [x])

- [x] I have updated the documentation, or
- [ ] No documentation update is required

## How I've tested my work (please select one)

I have verified these changes via:

- [x] Code inspection only, or
- [x] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [ ] Both unit tests + ran on a real repository

<!-- Do you have any suggestions about this PR template? Edit it here: https://github.com/renovatebot/renovate/edit/main/.github/pull_request_template.md -->

<!-- Please do not force push to your PR's branch after you have created your PR, as doing so forces us to review the whole PR again. This makes it harder for us to review your work because we don't know what has changed. -->
<!-- PRs will always be squashed by us when we merge your work. Commit as many times as you need in this branch. -->

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/renovatebot/renovate?pullRequest=37291) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/renovatebot/renovate?pullRequest=37291) before we can accept your contribution.<br/>**1** out of **2** committers have signed the CLA.<br/><br/>:white_check_mark: dominikfoerster2003<br/>:x: Dominik Förster<br/><hr/>**Dominik Förster** seems not to be a GitHub user. You need a GitHub account to be able to sign the CLA. If you have already a GitHub account, please [add the email address used for this commit to your account](https://help.github.com/articles/why-are-my-commits-linked-to-the-wrong-user/#commits-are-not-linked-to-any-user).<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/renovatebot/renovate?pullRequest=37291) it.</sub>

viceice reviewed (changes requested):
needs tests

nikaro commented on lib/modules/platform/types.ts:
```diff
@@ -107,6 +107,7 @@ export interface PlatformPrOptions {
   gitLabIgnoreApprovals?: boolean;
   usePlatformAutomerge?: boolean;
   forkModeDisallowMaintainerEdits?: boolean;
+  merge_after?: string;
```
Shouldn't this be in camelCase?

(Not a maintainer here, but just interested by this feature landing in Renovate.)

dominikfoerster2003 committed:
feat(gitlab): implement merge_after for automergeSchedule

Set the GitLab `merge_after` parameter on the automerge API request when `automergeSchedule` is configured and the current time is outside the schedule. This delays the actual merge until the next scheduled window, instead of relying solely on Renovate-side scheduling.

- Add `mergeAfter` to `PlatformPrOptions`
- Compute the next schedule time in `getPlatformPrOptions` via a new `getNextScheduleTime` helper that supports both Cron and later.js syntax with timezone awareness
- Pass the value through to the GitLab `/merge` endpoint as `merge_after` (snake_case for the API)
- Skip on the merge-trains code path, which doesn't support `merge_after`
- Add tests for `getNextScheduleTime`, `getPlatformPrOptions` and the new GitLab request body
